device_template/template.go

```diff
@@ -0,0 +1,65 @@
+package device_template
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"path"
+)
+
+const templateDir = "template_files"
+
+type FwVersion struct {
+	Type  string `json:"type"`
+	Major string `json:"major"`
+	Minor string `json:"minor"`
+}
+type Hardware struct {
+	Type    string `json:"type"`
+	Version struct {
+		Major string `json:"major"`
+		Minor string `json:"minor"`
+	} `json:"version"`
+}
+type DeviceTemplate struct {
+	FileName  string          `json:"file_name"`
+	FwVersion FwVersion       `json:"fw_version"`
+	Hardware  Hardware        `json:"hardware"`
+	Settings  json.RawMessage `json:"settings"`
+	Commands  json.RawMessage `json:"commands"`
+	Ports     json.RawMessage `json:"ports"`
+}
+
+var Templates = make(map[string]*DeviceTemplate)
+
+func init() {
+	scanTemplateDir()
+}
```

Comment:
lets not use init functions and global variables, this is tricky to deal with when we adding tests and when debugging.

implement a custom HTTP handler and do these when initializing it.

I remember you had some init functions and globals in other places so please remove these as well.